contracts/core/Dispatcher.sol

```diff
@@ -37,64 +51,26 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
     // keep track of outbound ack packets to prevent replay attack
     mapping(address => mapping(bytes32 => mapping(uint64 => bool))) ackPacketCommitment;
 
-    ConsensusStateManager consensusStateManager;
-
     //
     // methods
     //
-    constructor(string memory initPortPrefix, ConsensusStateManager _consensusStateManager) {
+    constructor(string memory initPortPrefix) {
         portPrefix = initPortPrefix;
         portPrefixLen = uint32(bytes(initPortPrefix).length);
-        consensusStateManager = _consensusStateManager;
     }
 
     //
     // Utility functions
     //
 
-    /**
-     * Convert a non-0x-prefixed hex string to an address
-     * @param hexStr hex string to convert to address. Note that the hex string must not include a 0x prefix.
-     * hexStr is case-insensitive.
-     */
-    function hexStrToAddress(string memory hexStr) internal pure returns (address) {
-        if (bytes(hexStr).length != 40) {
-            revert invalidHexStringLength();
-        }
-
-        bytes memory strBytes = bytes(hexStr);
-        bytes memory addrBytes = new bytes(20);
-
-        for (uint256 i = 0; i < 20; i++) {
-            uint8 high = uint8(strBytes[i * 2]);
-            uint8 low = uint8(strBytes[1 + i * 2]);
-            // Convert to lowercase if the character is in uppercase
-            if (high >= 65 && high <= 90) {
-                high += 32;
-            }
-            if (low >= 65 && low <= 90) {
-                low += 32;
-            }
-            uint8 digit = (high - (high >= 97 ? 87 : 48)) * 16 + (low - (low >= 97 ? 87 : 48));
-            addrBytes[i] = bytes1(digit);
-        }
-
-        address addr;
-        assembly {
-            addr := mload(add(addrBytes, 20))
-        }
-
-        return addr;
-    }
-
     // verify an EVM address matches an IBC portId.
     // IBC_PortID = portPrefix + address (hex string without 0x prefix, case-insensitive)
     function portIdAddressMatch(address addr, string calldata portId) public view returns (bool) {
         if (keccak256(abi.encodePacked(portPrefix)) != keccak256(abi.encodePacked(portId[0:portPrefixLen]))) {
             return false;
         }
         string memory portSuffix = portId[portPrefixLen:];
-        return hexStrToAddress(portSuffix) == addr;
+        return IbcUtils.hexStrToAddress(portSuffix) == addr;
```

Comment:
One of the reasons we didn't use libraries to their full extent is because we ran into troubles when generating Typescript bindings (https://github.com/open-ibc/vibc-core-smart-contracts/pull/41) 
At the time, the simplest solution was to avoid them in some places, that's why you'll see that the `Dispatcher` contract inherits from `Ibc` as a opposed to use it as a library.  In any case, I think libraries make the most sense and we should fix or workaround the binding issues. 

Just wanted to call this out in case you run into the same problems. And if so, now is the time to find a real fix.